feat: add `cursor_text_color`

wincent opened the pull request:
Cherry-pick of:

- https://github.com/kdrag0n/base16-kitty/pull/12

from the "original" [base16-kitty repo](https://github.com/kdrag0n/base16-kitty) (ie. the one [referenced](https://github.com/chriskempson/base16-templates-source/blob/394b12c904fa5d0b8cf458639877dcc484520194/list.yaml#L38) from [chriskempson/base16-templates-source](https://github.com/chriskempson/base16-templates-source)).

Original description:

> By default, Kitty is setting the cursor text color to `#111111`.
>
> See https://sw.kovidgoyal.net/kitty/conf/#opt-kitty.cursor_text_color.
>
> It's not rendering well when using vim with light theme.
>
> ![348778491-776f0881-fcf1-4e0b-8ff9-480cf32b3a2c](https://github.com/user-attachments/assets/4d6c6e86-32bc-4c1f-a039-9b3b03154cee)

l-lin committed:
Add cursor_text_color

By default, Kitty is setting the cursor text color to #111111.

See https://sw.kovidgoyal.net/kitty/conf/#opt-kitty.cursor_text_color.

It's not rendering well when using vim with light theme.

JamyGolden commented:
Thanks for the PR. Can you add this change to the `base24.mustache` and `base24-256.mustache` file?

wincent committed:
chore: replicate change in base24 mustache templates as well

As requested here:

- https://github.com/tinted-theming/tinted-kitty/pull/2#issuecomment-2378058791

> Can you add this change to the base24.mustache and base24-256.mustache
> file?

wincent commented:
> Thanks for the PR. Can you add this change to the `base24.mustache` and `base24-256.mustache` file?

Sure thing. Done in 4e9258dc8224810bcaede11ba0d13b4c049c16a9.

JamyGolden reviewed (approved):
Thanks!

JamyGolden commented:
I’ve just run a build so the themes have been updated.
